pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -335,6 +335,9 @@ func (c *Controller) syncClusterDeployment(key string) error {
 		if !hasFinalizer(clusterDeployment) {
 			return nil
 		}
+		if !hasSynced(clusterDeployment) {
+			return c.deleteFinalizer(clusterDeployment)
+		}
```

Comment:
Would it be better if we only added the finalizer after successfully syncing machine sets?

Reply:
I like that better... updated